File: Course 04 [Algorithms & Problem-Solving Level 1 Solutions]/Abu_HadHoud_Solutions_Using_C++/Problem_14.cpp
#include <iostream>
using namespace std;

void ReadNumbers(int&Num1 , int& Num2) {
	cout << "Please enter Number A : ";	
	cin >> Num1;

	cout << "Please enter Number B : ";
	cin >> Num2;
}
void SwapNumbers(int& A, int& B) {
	int Temp;
	Temp= A;
	A = B;
	B = Temp;
	 

}
void PrintNumbers(int Num1, int Num2) {
	cout << "\nNumber 1 =  " << Num1 << endl;
	cout << "Number 2 =  " << Num2 << endl;
}	
int main() {
	int Num1, Num2;
	ReadNumbers(Num1, Num2);
	PrintNumbers(Num1, Num2);
	SwapNumbers(Num1, Num2);
	PrintNumbers(Num1, Num2);



	return 0;
}

File: Course 04 [Algorithms & Problem-Solving Level 1 Solutions]/My_Solutions_Using_C#/Problem_14.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Net;
using System.Reflection.Metadata.Ecma335;
using System.Threading;
using System.Threading.Tasks;

namespace AbuHadHoud
{
    internal class Program
    {
        enum enPassFail { Pass = 1, Fail = 2 }
        static void ReadNumber(out int Num1, out int Num2)
        {
            Console.Write("Please enter Number 1 : ");
            Num1 = int.Parse(Console.ReadLine());

            Console.Write("Please enter Number 2 : ");
            Num2 = int.Parse(Console.ReadLine());
        }

        static void Swap(ref int Num1, ref int Num2)
        {
            int Temp = Num1;
            Num1 = Num2;
            Num2 = Temp;

        }


        static void PrintResult(int Num1 , int Num2)
        {
            Console.WriteLine("");
            Console.WriteLine($"Number 1 = {Num1}");
            Console.WriteLine($"Number 2 = {Num2}");

        }
        static void Main(string[] args)

        {
            int Num1, Num2;
            ReadNumber(out Num1, out Num2);
            PrintResult(Num1, Num2);
            Swap(ref Num1, ref Num2);
            PrintResult(Num1, Num2);

        }
    }
}